jdaviz/configs/default/plugins/virtual_observatory/vo_plugin.py

```diff
@@ -0,0 +1,419 @@
+            # source coordinate and add a Spatial search constraint
+            if self.resource_filter_coverage:
+                try:
+                    # First parse user-provided source as direct coordinates
+                    coord = SkyCoord(
+                        self.source, unit=u.deg, frame=self.coordframe_selected
+                    )
+                except Exception:
+                    try:
+                        # If that didn't work, try parsing it as an object name
+                        coord = SkyCoord.from_name(
+                            self.source, frame=self.coordframe_selected
+                        )
+                    except Exception:
+                        self.resources_loading = False  # Stop loading bar
+                        self.hub.broadcast(
+                            SnackbarMessage(
+                                f"Unable to resolve source coordinates: {self.source}",
+                                sender=self,
+                                color="error",
+                            )
+                        )
+                        raise LookupError(
+                            f"Unable to resolve source coordinates: {self.source}"
+                        )
+                registry_args.append(
+                    registry.Spatial(
+                        (coord, (self.radius_val * u.Unit(self.radius_unit.selected))),
+                        intersect="overlaps",
+                    )
+                )
+            self._full_registry_results = registry.search(*registry_args)
+            self.resources = list(self._full_registry_results.getcolumn("short_name"))
+        except DALFormatError as e:
+            if type(e.cause) is RequestConnectionError:
+                self.hub.broadcast(
+                    SnackbarMessage(
+                        f"Can't connect to VO registry. Check your internet connection: {e}",
+                        sender=self,
+                        color="error",
+                    )
+                )
+            else:
+                raise e
+        except Exception as e:
+            self.hub.broadcast(
+                SnackbarMessage(
+                    f"An error occured querying the VO Registry: {e}",
+                    sender=self,
+                    color="error",
+                )
+            )
+            raise
+        finally:
+            self.resources_loading = False  # Stop loading bar
+
+    def vue_query_resource(self, _=None):
+        """
+        Once a specific VO resource is selected, query it with the user-specified source target.
+        User input for source is first attempted to be parsed as a SkyCoord coordinate. If not,
+        then attempts to parse as a target name.
+        """
+        # Reset Table
+        self.table.items = []
+        self._populate_url_only = False
+        self.table.headers_visible = ["Title", "Instrument", "DateObs"]
```

Comment:
should this wait until we know it will be successful?  Otherwise any exception will leave the table reset (or maybe that makes sense).

Reply:
The URL is one of the only guaranteed columns that will return; the other columns are mostly standardized, but there are, and will always be, exceptions. I felt like metadata enforcement shouldn't prevent the behavior of the plugin, especially if there's only one result, so this seemed like the most sensible way to ensure the plugin continued to function while not getting bogged down by other "weirdness" in the metadata. It's hard to explicitly catch what errors will come up in any case; there are some weird ones you wouldn't expect to come up, like this `nan` issue in the CANDELS survey Cami found: https://github.com/astropy/pyvo/pull/550

I guess in many words, yes, the idea was that any exception in the metadata wouldn't prevent the results from being showed, short of the URL being missing, which would be a proper error.  If you feel otherwise, let me know!